File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/WindowElements/adddynamicgridsandconnectionpoints.rb
module Asposediagramjava
  module AddDynamicGridsAndConnectionPoints
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # get window object by index
        window = diagram.getWindows().get(0)
        
        # check dynamic grid option
        window.setDynamicGridEnabled(1)
        
        # check connection points option
        window.setShowConnectionPoints(1)

        # Save as VDX
        diagram.save(data_dir + "AddDynamicGridsAndConnectionPoints.vsx", Rjb::import('com.aspose.diagram.SaveFileFormat').VSX)

        puts "Added Support of Dynamic Grids and Connection Points in the Visio Drawings."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithWindowElements/AddDynamicGridsAndConnectionPoints.php
<?php
namespace Aspose\Diagram\WorkingwithWindowElements;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class AddDynamicGridsAndConnectionPoints{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."Drawing.vsd");

        # get window object by index
        $window=$diagram->getWindows()->get(0);

        # check dynamic grid option
        $window->setDynamicGridEnabled(1);

        # check connection points option
        $window->setShowConnectionPoints(1);

        # Save as VDX
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."AddDynamicGridsAndConnectionPoints.vsx", $saveFileFormat->VSX);

        print "Added Support of Dynamic Grids and Connection Points in the Visio Drawings.".PHP_EOL;
    }
}
